gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/HiveProxyQueryExecutor.java

```diff
@@ -0,0 +1,223 @@
+          connectionMap.put(proxy, hiveConnection);
+          for (String setting : settings) {
+            statement.execute(setting);
+          }
+        }
+        return null;
+      }
+    });
+  }
+
+  private synchronized void setConnection()
+      throws ClassNotFoundException, SQLException {
+    HiveConnection hiveConnection = getHiveConnection();
+    Statement statement = hiveConnection.createStatement();
+    this.statementMap.put(DEFAULT, statement);
+    this.connectionMap.put(DEFAULT, hiveConnection);
+    for (String setting : settings) {
+      statement.execute(setting);
+    }
+  }
+
+  private HiveConnection getHiveConnection(String proxyUser)
+      throws ClassNotFoundException, SQLException {
+    Class.forName("org.apache.hive.jdbc.HiveDriver");
+    Preconditions.checkArgument(this.state.contains(HIVE_JDBC_URL), "Missing required property " + HIVE_JDBC_URL);
+    String url = this.state.getProp(HIVE_JDBC_URL);
+    url = url + HIVE_SERVER2_PROXY_USER + proxyUser;
+    return (HiveConnection) DriverManager.getConnection(url);
+  }
+
+  private HiveConnection getHiveConnection()
+      throws ClassNotFoundException, SQLException {
+    Class.forName("org.apache.hive.jdbc.HiveDriver");
+    Preconditions.checkArgument(this.state.contains(HIVE_JDBC_URL), "Missing required property " + HIVE_JDBC_URL);
+    String url = this.state.getProp(HIVE_JDBC_URL);
+    return (HiveConnection) DriverManager.getConnection(url);
+  }
+
+  @Override
+  public void executeQueries(List<String> queries)
+      throws SQLException {
+    Optional<String> proxy = Optional.absent();
+    executeQueries(queries, proxy);
+  }
+
+  @Override
+  public void executeQuery(String query)
+      throws SQLException {
+    Optional<String> proxy = Optional.absent();
+    executeQuery(query, proxy);
+  }
+
+  public void executeQueries(List<String> queries, Optional<String> proxy)
+      throws SQLException {
+    Preconditions.checkArgument(!this.statementMap.isEmpty(), "No hive connection. Unable to execute queries");
+    if (!proxy.isPresent()) {
+      Preconditions.checkArgument(this.statementMap.size() == 1, "Multiple Hive connections. Please specify a user");
+      proxy = Optional.fromNullable(this.statementMap.keySet().iterator().next());
+    }
+    Statement statement = this.statementMap.get(proxy.get());
+    for (String query : queries) {
+      statement.execute(query);
+    }
+  }
+
+  public void executeQuery(String query, Optional<String> proxy)
+      throws SQLException {
+    Preconditions.checkArgument(!this.statementMap.isEmpty(), "No hive connection. Unable to execute queries");
+    if (!proxy.isPresent()) {
+      Preconditions.checkArgument(this.statementMap.size() == 1, "Multiple Hive connections. Please specify a user");
+      proxy = Optional.fromNullable(this.statementMap.keySet().iterator().next());
+    }
+    Statement statement = this.statementMap.get(proxy.get());
+    for (String setting : this.settings) {
```

Comment:
Why `executeQuery` is checking `this.settings` while `executeQueries` is not?
Plus, these two methods share many duplicate codes. You can call `executeQueries(Lists.newArrayList(query), proxy)` in `executeQuery`.